packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -1,26 +1,41 @@
+const report = require(`gatsby-cli/lib/reporter`)
 const axios = require(`axios`)
 const _ = require(`lodash`)
 const { createRemoteFileNode } = require(`gatsby-source-filesystem`)
 const { URL } = require(`url`)
 const { nodeFromData } = require(`./normalize`)
+const asyncPool = require(`tiny-async-pool`)
 
 exports.sourceNodes = async (
   { actions, store, cache, createNodeId, createContentDigest },
-  { baseUrl, apiBase, basicAuth, filters, headers, params }
+  {
+    baseUrl,
+    apiBase,
+    basicAuth,
+    filters,
+    headers,
+    params,
+    concurrentFileRequests,
+  }
 ) => {
   const { createNode } = actions
+  const drupalFetchActivity = report.activityTimer(`Fetch data from Drupal`)
+  const downloadingFilesActivity = report.activityTimer(`Remote file download`)
 
   // Default apiBase to `jsonapi`
   apiBase = apiBase || `jsonapi`
 
+  // Default concurrentFileRequests to `20`
+  concurrentFileRequests = concurrentFileRequests || 20
```

Comment:
could you explain why 20 is the magic number? 😄

Reply:
It's dependent on your pipeline's or instance's memory allocation.

20 should be sufficient for all cases. This, of course, can be changed if you know you have sufficient memory to handle the number of promises in the call-stack.

Oddly, in RL testing I found the image size would also affect the success of the runtime if the concurrency pool size was too large.